packages/secure_storage/amplify_secure_storage_dart/lib/src/platforms/amplify_secure_storage_web.dart

```diff
@@ -14,16 +14,66 @@
 
 import 'dart:async';
 
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
 import 'package:amplify_secure_storage_dart/src/exception/not_available_exception.dart';
 import 'package:amplify_secure_storage_dart/src/exception/secure_storage_exception.dart';
-import 'package:amplify_secure_storage_dart/src/interfaces/amplify_secure_storage_interface.dart';
-import 'package:amplify_secure_storage_dart/src/interfaces/secure_storage_interface.dart';
 import 'package:amplify_secure_storage_dart/src/js/indexed_db.dart';
+import 'package:amplify_secure_storage_dart/src/platforms/amplify_secure_storage_in_memory.dart';
 
 /// The web implementation of [SecureStorageInterface].
 class AmplifySecureStorageWeb extends AmplifySecureStorageInterface {
   AmplifySecureStorageWeb({required super.config});
 
+  /// The [SecureStorageInterface] instance to use.
+  late final Future<SecureStorageInterface> _instance = () async {
+    if (config.webOptions.persistenceOption == WebPersistenceOption.inMemory) {
+      return const AmplifySecureStorageInMemory();
+    }
+    if (indexedDB == null) {
+      logger.warn(
+        'IndexedDB is not available. '
+        'Falling back on in-memory storage.',
+      );
+      return const AmplifySecureStorageInMemory();
+    }
+    // indexedDB will be non-null in Firefox private browsing,
+    // but will fail to open.
+    try {
+      final openRequest = indexedDB!.open('test', 1);
+      await openRequest.future;
+      return _IndexedDBStorage(config: config);
+    } on Object catch (_) {
+      logger.warn(
+        'Could not open IndexedDB database. '
+        'It may not be supported by the current browser. '
+        'Falling back on in-memory storage.',
+      );
```

Comment:
If you think it's helpful to include, you could do this too :

```suggestion
    } on Object catch (e) {
      logger.warn(
        'Could not open IndexedDB database. '
        'It may not be supported by the current browser. '
        'Falling back on in-memory storage.',
        e,
      );
```

I generally prefer not catching anything if it's unused, though

```suggestion
    } on Object {
      logger.warn(
        'Could not open IndexedDB database. '
        'It may not be supported by the current browser. '
        'Falling back on in-memory storage.',
      );
```